universal-application-tool-0.0.1/app/views/admin/versions/VersionListView.java

```diff
@@ -0,0 +1,175 @@
+package views.admin.versions;
+
+import static j2html.TagCreator.body;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.each;
+import static j2html.TagCreator.p;
+import static j2html.TagCreator.table;
+import static j2html.TagCreator.tbody;
+import static j2html.TagCreator.td;
+import static j2html.TagCreator.th;
+import static j2html.TagCreator.thead;
+import static j2html.TagCreator.tr;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import controllers.admin.routes;
+import j2html.tags.ContainerTag;
+import j2html.tags.Tag;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import models.LifecycleStage;
+import models.Version;
+import play.mvc.Http;
+import play.twirl.api.Content;
+import views.BaseHtmlView;
+import views.admin.AdminLayout;
+import views.components.LinkElement;
+import views.style.BaseStyles;
+import views.style.ReferenceClasses;
+import views.style.StyleUtils;
+import views.style.Styles;
+
+public class VersionListView extends BaseHtmlView {
+
+  private final AdminLayout layout;
+
+  @Inject
+  public VersionListView(AdminLayout layout) {
+    this.layout = layout;
+  }
+
+  public Content render(List<Version> allVersions, Http.Request request) {
+    Optional<Version> draftVersion =
+        allVersions.stream()
+            .filter(version -> version.getLifecycleStage().equals(LifecycleStage.DRAFT))
+            .findAny();
+    Optional<Version> activeVersion =
+        allVersions.stream()
+            .filter(version -> version.getLifecycleStage().equals(LifecycleStage.ACTIVE))
+            .findAny();
+    ImmutableList<Version> olderVersions =
+        allVersions.stream()
+            .filter(version -> version.getLifecycleStage().equals(LifecycleStage.OBSOLETE))
+            .collect(ImmutableList.toImmutableList());
+    ContainerTag bodyContent =
+        body(
+            renderHeader("Current Versions"),
+            renderVersionCard(draftVersion),
+            renderVersionCard(activeVersion),
+            renderHeader("Older Versions"),
+            renderPastVersionTable(olderVersions, request));
+
+    return layout.render(bodyContent);
+  }
+
+  private String renderDateTime(Instant time) {
+    LocalDateTime datetime = LocalDateTime.ofInstant(time, ZoneId.of("America/Los_Angeles"));
+    return datetime.format(DateTimeFormatter.ofPattern("yyyy/MM/dd 'at' h:mm a"));
```

Comment:
Suggestion (blocking): extract `DateTimeFormatter.ofPattern("yyyy/MM/dd 'at' h:mm a")` to a shared constant on `BaseHtmlView`

Reply:
Er, actually please exact this method instead, sorry.

Reply:
Done.